src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/LegacyRazorCompletionEndpoint.cs

```diff
@@ -145,6 +145,11 @@ internal class LegacyRazorCompletionEndpoint : IVSCompletionEndpoint
             _logger.LogTrace($"Resolved {razorCompletionItems.Count} completion items.");
 
             var completionList = CreateLSPCompletionList(razorCompletionItems);
+            var completionCapability = _clientCapabilities?.TextDocument?.Completion as VSInternalCompletionSetting;
+
+            // The completion items are cached and can be retrieved via this result id to enable the "resolve" completion functionality.
+            var resultId = _completionListCache.Set(completionList, razorCompletionItems);
```

Comment:
Since we changed the completion list cache type I also had to update the legacy endpoint to work. I validated tests & end-to-end functionality wasn't disrupted.